src/views/templateVariables.ts

```diff
@@ -0,0 +1,97 @@
+import joplin from "api";
+import { encode } from "html-entities";
+
+const getVariableHtml = (variable: string, type: string): string => {
+    type = type.trim();
+
+    if (type == "text") {
+        return (
+            `
+            <div class="variableInput">
+                <div class="variableName">
+                    ${encode(variable)}
+                </div>
+                <div>
+                    <input name="${encode(variable)}" type="text"></input>
+                </div>
+            </div>
+            `
+        );
+    }
+
+    if (type == "number") {
+        return (
+            `
+            <div class="variableInput">
+                <div class="variableName">
+                    ${encode(variable)}
+                </div>
+                <div>
+                    <input name="${encode(variable)}" type="number"></input>
+                </div>
+            </div>
+            `
+        );
+    }
+
+    if (type == "boolean") {
+        return (
+            `
+            <div class="variableInput">
+                <div class="variableName">
+                    ${encode(variable)}
+                </div>
+                <div>
+                    <select name="${encode(variable)}">
+                        <option value="true">Yes</option>
+                        <option value="false">No</option>
+                    </select>
+                </div>
+            </div>
+            `
+        );
+    }
+
+    if (type.startsWith("enum(") && type.endsWith(")")) {
```

Comment:
I wonder if it makes more sense to use the yaml list syntax for this? I think this will affect how your parser works, but should be a lot better for users. For example
This
```yaml
fruit: enum(apple, orange, banana)
```

Would become this
```yaml
fruit:
  - apple
  - orange
  - banana
```

Reply:
This isn't a requirement for this pull request, but it's something you should think about and perhaps bring to the community.

Reply:
Hmm, in terms of implementation I don't think it's hard to implement. But the way the feature is currently implemented it will create some inconsistency in case we want to add more type of variables in future.

In terms of UX, I want to understand the reason you said

> but should be a lot better for users

Because according to me, the users will have to do an equal amount of work in both cases they will have to look to the plugin documentation to see how should they declare enum template variables since it's not a standard.

But, I agree on the fact that as it's something like a list it seems more semantically correct to declare the options in the list syntax.

What do you think?

Reply:
Actually, I have another idea. Let's create an issue to keep track of this. And in the future, we can support both these syntaxes.

Reply:
> What do you think?

My worry is that most users have never heard of an enum, so it's nonsense to them. But a list will make sense to them.

Tracking this in a seperate issue is a good idea, I'm looking forward to hearing your ideas.

Reply:
Ah, okay. Got it, I always forget that a large fraction of Joplin users are non-programmers and enum is kind of a technical word. Sure, will create an issue for this.